integration/spark/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -143,8 +144,12 @@ object DataLoadProcessBuilderOnSpark {
 
     var numPartitions = CarbonDataProcessorUtil.getGlobalSortPartitions(
       configuration.getDataLoadProperty(CarbonCommonConstants.LOAD_GLOBAL_SORT_PARTITIONS))
+
+    // if numPartitions user does not specify and not specified in config then dynamically calculate
     if (numPartitions <= 0) {
-      numPartitions = convertRDD.partitions.length
+      val defaultMaxSplitBytes = sessionState(sparkSession).conf.filesMaxPartitionBytes
```

Comment:
`filesMaxPartitionBytes` actually gives the parquet block size, do we need to make it configurable? like carbon block size or something like that. what do you say @QiangCai